shell/platform/embedder/embedder.h

```diff
@@ -943,6 +943,11 @@ typedef struct {
   VoidCallback destruction_callback;
 } FlutterSoftwareBackingStore;
 
+typedef struct {
+  // A Metal texture for Flutter to render into.
+  FlutterMetalTexture texture;
```

Comment:
Are textures the only way to specify render targets to Metal. I though `MTLDrawable` was the protocol. In case there are multiple ways to specify these render target, we might need to make this a tagged union similar to `FlutterOpenGLBackingStore`.

Reply:
This texture is an Objective-C object. For these, the ownership rules must be clearly documented. Is the texture transfer-in or transfer-none? I prefer the latter because the engine can retain it explicitly and early returns in case of error don't affect how the embedder must behave.